### PR TITLE
add TorchAPI for SpaceEngineers

### LINKED ISSUE #1349: [Egg Request]: Space engineers Torch
### Service

Space engineers Torch

### Does this expand an already existing service

Yes

### Link to game

https://store.steampowered.com/app/244850/Space_Engineers/

### Links for server downloads

https://torchapi.net/download

### Links for the install docs

https://wiki.torchapi.net/index.php/Basic_Getting_Started_Guide

### PR DESCRIPTION
**Pull request by gOOvER**
# Description

Torch is the successor to SE Server Extender and gives server admins the tools they need to keep their Space Engineers servers running smoothly. It features a user interface with live management tools and a plugin system so you can run your server exactly how you'd like. Torch is still in early development so there may be bugs and incomplete features.

## Changelog

- add TorchAPI Egg
- fix SE Egg for new folderstructure
- disabled to add own Worlds, because own world make problems
- readded configparsing for world config
- changed Author of the SE Egg, because Egg is now complete rewritten
- fixed a lot of Spelling typos 

fixes #1349 

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
3. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?

### CODE REVIEW

File: game_eggs/steamcmd_servers/space_engineers/torch/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+**NOTE**
+Because of the winetricks installation, the install process could take some time. Likewise, the first start of the server takes its time.
+
+When installed, the Server loads a base World and config from this repo. New world must be generated with the Windows Dedicated Server and then uploaded to the /config/Saves folder.
+Everything which is not defined in a variable could be edited directy in the config file /config/SpaceEngineers-Dedicated.cfg and config/Saves/{{WORLD}}/Sandbox.sbc
+
+## Server Ports
+
+Default server ports are listed below, but the Main, STEAm and API port can be any port.
```
Comment (Cryotize) [inline game_eggs/steamcmd_servers/space_engineers/torch/README.md:20]:
```suggestion
Default server ports are listed below, but the Main, Steam and API port can be any port. 
```

File: game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            "description": "Steam Port",
+            "env_variable": "STEAM_PORT",
+            "default_value": "8766",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|numeric|max:65535",
+            "field_type": "text"
+        },
+        {
+            "name": "Enable Remote Api",
```
Comment (Cryotize) [inline game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json:124]:
```suggestion
            "name": "Enable Remote API",
```

File: game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            "description": "Enable Remote Api",
```
Comment (Cryotize) [inline game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json:125]:
```suggestion
            "description": "Enable Remote API",
```

File: game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            "env_variable": "REMOTEAPI_ENABLE",
+            "default_value": "false",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|in:true,false",
+            "field_type": "text"
+        },
+        {
+            "name": "Remote Api Port",
```
Comment (Cryotize) [inline game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json:134]:
```suggestion
            "name": "Remote API Port",
```

File: game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+            "description": "Remote Api Port",
```
Comment (Cryotize) [inline game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json:135]:
```suggestion
            "description": "Remote API Port",
```

File: game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+            "user_viewable": false,
+            "user_editable": false,
+            "rules": "required|numeric|digits_between:1,6",
+            "field_type": "text"
+        },
+        {
+            "name": "Server Name",
+            "description": "Name of the server, appears in Steam browser.",
+            "env_variable": "SERVER_NAME",
+            "default_value": "Pterodactyl Space Engineer Server",
```
Comment (Cryotize) [inline game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json:47]:
```suggestion
            "default_value": "A Pterodactyl Hosted Space Engineer Server",
```

File: game_eggs/steamcmd_servers/space_engineers/torch/README.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+Default server ports are listed below, but the Main, STEAm and API port can be any port. 
+
+| Port  | default |
+|-------|---------|
+| Game  | 27016   |
+| Steam | 8766    |
+| API   | 8081    |
+
+## Minumum server settings
```
Comment (Cryotize) [inline game_eggs/steamcmd_servers/space_engineers/torch/README.md:28]:
```suggestion
## Minimum server settings
```

**Review by Cryotize — COMMENTED** [review]
Added some minor changes and corrections. 

I'm not sure if this kind of server is desired though, cause Pterodactyl itself is a management panel.

File: game_eggs/steamcmd_servers/space_engineers/torch/README.md (hunk 1 above)
Comment (gOOvER) [reply game_eggs/steamcmd_servers/space_engineers/torch/README.md:20]: only Copy and Paste from their site

**Comment by gOOvER** [comment]
Funny, you reviewd this, but not the Defaultegg

**Comment by gOOvER** [comment]
> Added some minor changes and corrections. 
> 
> I'm not sure if this kind of server is desired though, cause Pterodactyl itself is a management panel. 

Did you test the Egg or is this more a "Klugscheißen" ?

File: game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json (hunk 6 above)
Comment (gOOvER) [reply game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json:47]: Not needed

**Comment by Cryotize** [comment]
> Did you test the Egg or do you really know, what torch is? Looks Like not. Sounds more like a Know-it-all. And btw, its not my first Egg. ;)
> 
> i will fix the spelling mistakes as soon as i am at home. however, these are already present since the first appearance of the egg

I'm sorry for seeming like a "Klugscheisser", this was not the goal of my review. Also, no i do not know Torch or SpaceEngineers at all and did not test the Egg. I just checked out your profile due to your comment on my PR and saw your commit history.

I just try to help where i can, in this case spelling mistakes due to my limited technical knowledge.

> I'm not sure if this kind of server is desired though, cause Pterodactyl itself is a management panel.

Sorry if this seems like a "Know-it-All" comment to you, I was going trough closed PRs from ARK: Survival Evolved, where someone requested something similar and it got denied. This is why i added my thought about a separate management panel (as of my understanding and their description).

Sorry again and i hope that i could still help :)

File: game_eggs/steamcmd_servers/space_engineers/torch/README.md (hunk 1 above)
Comment (Cryotize) [reply game_eggs/steamcmd_servers/space_engineers/torch/README.md:20]: The idea behind the change from "Main" to "Game" was due to the table with the required ports. There they are called "Game, Steam and API"

**Comment by S1L3NTANG3L** [comment]
Fix suggested by @gOOvER works, link [https://haste.goover.de/edeyocumiy.xml](https://haste.goover.de/edeyocumiy.xml)

File: game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+            "container": "debian:bullseye-slim",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json:28]: This is a steam game is there any reason this is not using the debian installer image so the apt install and update are not needed?

File: game_eggs/steamcmd_servers/space_engineers/default/SpaceEngineers-Dedicated.cfg [hunk 9]
```diff
@@ -1,3 +1,5 @@
+
+Save New Duplicate & Edit Just Text Twitter
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/space_engineers/default/SpaceEngineers-Dedicated.cfg:2]: Does this need to be here?

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/space_engineers/default/SpaceEngineers-Dedicated.cfg (hunk 9 above)
Comment (gOOvER) [reply game_eggs/steamcmd_servers/space_engineers/default/SpaceEngineers-Dedicated.cfg:2]: Funny :D idk why its there :D

File: game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json (hunk 8 above)
Comment (gOOvER) [reply game_eggs/steamcmd_servers/space_engineers/torch/egg-space-engineers--torch-server.json]: Oh, i missed that. The image was neede in past for Wine install

**Review by QuintenQVD0 — APPROVED** [review]

**Pull request merged**